src/tpm2_wrap.c

```diff
@@ -2539,20 +2535,35 @@ int wolfTPM2_SignHashScheme(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* key,
 
     if (key->pub.publicArea.type == TPM_ALG_ECC) {
         /* Assemble R and S into signature (R then S) */
-        *sigSz = signOut.signature.signature.ecdsa.signatureR.size +
-                signOut.signature.signature.ecdsa.signatureS.size;
-        XMEMCPY(sig, signOut.signature.signature.ecdsa.signatureR.buffer,
-            signOut.signature.signature.ecdsa.signatureR.size);
-        XMEMCPY(sig + signOut.signature.signature.ecdsa.signatureR.size,
-            signOut.signature.signature.ecdsa.signatureS.buffer,
-            signOut.signature.signature.ecdsa.signatureS.size);
+        sigOutSz = signOut.signature.signature.ecdsa.signatureR.size +
```

Comment:
Should there be a check here that R and S sizes are equal?   You assume so below during the memcpy()